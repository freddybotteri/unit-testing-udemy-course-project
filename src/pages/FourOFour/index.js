import React, { Component } from "react";

class FourOFour extends Component {
  render() {
    return(<div/>)
  }
}

export default FourOFour;
