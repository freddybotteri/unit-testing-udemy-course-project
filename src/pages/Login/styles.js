export const styles = {
  container: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    height: "100vh",
    overflow: "hidden",
    width: "100%"
  },
  card: {
    width: "400px",
    minWidth: "300px",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center"
  },
  title: {
    marginTop: 12,
    fontSize: 18,
    marginBottom: 4
  }
}
