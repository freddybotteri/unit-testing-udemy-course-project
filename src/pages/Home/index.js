import React, { Component } from "react";
import PropTypes from "prop-types";
import Grid from "@material-ui/core/Grid";
import { withStyles } from "@material-ui/core/styles";
import { compose } from "redux";
import { connect } from "react-redux";
import get from "lodash/get";

import { getProfileData, getProfileRepos } from "../../redux/actions/profile";
import LeftContainer from "./components/LeftContainer";
import RightContainer from "./components/RightContainer";
import { styles } from "./style";
import { getData, clearStorage } from "../../utils/storage";
import { GITHUB_TOKEN } from "../../consts";

class Home extends Component {
  constructor() {
    super();
    this.state = { githubToken: null };
  }

  async componentDidMount() {
    const githubToken = getData(GITHUB_TOKEN);
    await this.setState({ githubToken });
    this.props.getProfileData({ githubToken });
  }

  componentDidUpdate(prevProps, prevState) {
    const { githubData, getProfileRepos } = this.props;
    const { githubToken } = this.state;
    const reposURL = get(githubData, "repos_url", null);
    const prevReposURL = get(prevProps, "githubData.repos_url", null);

    if (reposURL && prevReposURL !== reposURL) {
      getProfileRepos({ githubToken, reposURL });
    }
  }

  handleLogoutClick = () => {
    clearStorage();
    window.location.reload();
  };

  render() {
    const { classes, githubData, githubUserRepos } = this.props;

    return (
      <Grid container className={classes.homeContainer}>
        <Grid item xs={3} className={classes.leftContainer}>
          <LeftContainer {...githubData} onLogout={this.handleLogoutClick} />
        </Grid>
        <Grid item xs={9} className={classes.rightContainer}>
          <RightContainer repos={githubUserRepos} />
        </Grid>
      </Grid>
    );
  }
}

const actions = {
  getProfileData,
  getProfileRepos
};

const mapStateToProps = state => ({
  githubData: get(state, "profile.githubData", null),
  githubUserRepos: get(state, "profile.userRepos", null)
});

export { Home };

export default compose(
  connect(
    mapStateToProps,
    actions
  ),
  withStyles(styles)
)(Home);
