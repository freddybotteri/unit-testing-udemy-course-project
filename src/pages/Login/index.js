import React, { Component } from "react";
import { withStyles } from '@material-ui/core/styles';
import Typography from "@material-ui/core/Typography";
import Card from "@material-ui/core/Card";
import GithubLogin from "react-github-login";
import { connect } from "react-redux";
import { compose } from "redux";
import { Redirect} from "react-router-dom";

import { getGithubToken } from "../../redux/actions/login";
import { styles } from "./styles";

class Login extends Component {
  onSuccess = response => {
    if (response && response.code) {
      this.props.getGithubToken({ code: response.code });
    }
  }

  onFailure = response => {
    alert("No ha sido exitoso intenta de nuevo mas tarde!");
  }

  render() {
    const { classes, githubToken } = this.props;

    if (githubToken) {
      return <Redirect to="/home"/>;
    }

    return(
      <div className={classes.container}>
        <Card className={classes.card}>
          <Typography variant="h1" className={classes.title}>Iniciar sesion</Typography>
          <GithubLogin
            clientId={process.env.REACT_APP_GITHUB_CLIENT_ID}
            redirectUri=""
            className={classes.button}
            onSuccess={this.onSuccess}
            onFailure={this.onFailure}
            buttonText="Github account"/>
        </Card>
      </div>
    )
  }
}

const actions = {
  getGithubToken
};

const mapStateToProps = state => ({
  githubToken: state.login.githubToken
});

export { Login };

export default compose(
  withStyles(styles),
  connect(mapStateToProps, actions)
)(Login);
