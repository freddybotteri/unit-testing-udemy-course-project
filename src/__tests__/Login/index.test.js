import React from "react";
import { shallow } from "enzyme";
import Card from "@material-ui/core/Card";
import TestRenderer from "react-test-renderer";

import { Login } from "../../pages/Login";

let wrapper;

describe("<Login/>", () => {
  beforeEach(() => {
    const props = {
      classes: {},
      getGithubToken: jest.fn()
    };

    wrapper = shallow(<Login {...props} />);
  });

  it("Renderea sin error", () => {
    expect(wrapper).toBeDefined();
  });

  it("Renderea un Card", () => {
    expect(wrapper.find(Card)).toHaveLength(1);
  });

  it("Renderea correctamente las props", () => {
    expect(wrapper.instance().props.classes).toBeDefined();
    expect(wrapper.instance().props.classes).toEqual({});
    expect(wrapper.instance().props.getGithubToken).toBeDefined();
  });

  it("onSuccess test con response valido", () => {
    const response = { code: 12345 };

    wrapper.instance().onSuccess(response);
    expect(wrapper.instance().props.getGithubToken).toHaveBeenCalled();
  });

  it("onSuccess test con response invalido", () => {
    wrapper.instance().onSuccess();
    expect(wrapper.instance().props.getGithubToken).toHaveBeenCalledTimes(0);
  });

  it("Muestra una alerta, si la respuesta de github es erronea", () => {
    window.alert = jest.fn();
    wrapper.instance().onFailure({ error: "Hey" });
    expect(alert).toHaveBeenCalled();
    expect(alert).toHaveBeenCalledTimes(1);
  });

  it("El snapshot es correcto", () => {
    const props = {
      classes: {
        container: "container"
      },
      getGithubToken: jest.fn()
    };

    const tree = TestRenderer.create(<Login {...props} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
