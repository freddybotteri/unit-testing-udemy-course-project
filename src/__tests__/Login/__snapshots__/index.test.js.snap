// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> El snapshot es correcto 1`] = `
<div
  className="container"
>
  <div
    className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
  >
    <h1
      className="MuiTypography-root-29 MuiTypography-h1-41"
    >
      Iniciar sesion
    </h1>
    <button
      onClick={[Function]}
    >
      Github account
    </button>
  </div>
</div>
`;
